exonum/src/runtime/rust/service.rs

```diff
@@ -183,22 +195,52 @@ pub struct BeforeCommitContext<'a> {
     /// the database made by the previous transactions already executed in this block.
     pub fork: &'a Fork,
     /// Reference to the underlying runtime dispatcher.
-    dispatcher: &'a Dispatcher,
+    dispatcher: &'a DispatcherRef<'a>,
 }
 
 impl<'a> BeforeCommitContext<'a> {
     /// Create a new `BeforeCommit` context.
     pub(crate) fn new(
         instance: InstanceDescriptor<'a>,
         fork: &'a Fork,
-        dispatcher: &'a Dispatcher,
+        dispatcher: &'a DispatcherRef<'a>,
     ) -> Self {
         Self {
             instance,
             fork,
             dispatcher,
         }
     }
+
+    // TODO This method is hidden until it is fully tested in next releases. [ECR-3493]
+    #[doc(hidden)]
+    /// Create a client to call interface methods of the specified service instance.
+    pub fn interface<T>(&self, called: InstanceId) -> T
+    where
+        T: From<CallContext<'a>>,
+    {
+        self.call_context(called).into()
+    }
+
+    // TODO This method is hidden until it is fully tested in next releases. [ECR-3493]
+    #[doc(hidden)]
+    /// Create a context to call interfaces of the specified service instance.
+    pub fn call_context(&self, called: InstanceId) -> CallContext<'a> {
+        CallContext::new(self.fork, self.dispatcher, self.instance.id, called)
+    }
+
+    /// Add a configuration update to pending actions. These changes will be applied immediately
+    /// before the block commit.
+    ///
+    /// Only the supervisor service is allowed to perform this action.
+    #[doc(hidden)]
+    pub fn update_config(&self, changes: Vec<ConfigChange>) {
```

Comment:
If only supervisor is allowed to perform this method, why make it a part of the interface for all services?

Reply:
Because I can't distinguish at runtime supervisor from the other services.